### DISCUSSION TITLE
Closing a cancelled dialog window

### DISCUSSION THREAD

**Discussion by norricorp** [slint-ui/slint#4740] [Q&A]
In the example of a dialog box, "no" or "cancel" has no callback associated with it. I would expect "cancel" to close the dialog.
I saw discussion #3562 and that suggetsed using hide but is that going to end that function or will the dialog still be around but just not visible?
Is there a close function?

**Comment by norricorp** [comment]
The code I have used is 
	dialog_window.on_close_dialog({
		let dialog_window_handle = dialog_window.as_weak();
		move || {  
			dialog_window_handle.unwrap().window().hide().unwrap();  
		}
	});

and in slint!
StandardButton { kind: cancel; clicked => { root.close-dialog(); } }

This works but is it the right way?

**Reply by tronical** [reply]
Yes, that looks correct to me :)

**Comment by norricorp** [comment]
For the OK button, I added two functions, the last of which hides the dialog which works. But if the dialog is reopened then any text entered in the line edit field is still there. So hide is hide, not close. I added an extra line before the hide to clear the text after processing.

Would a close function be useful on the dialog window?
